### ISSUE TITLE
this plugin destroy sourcemap.

### ISSUE THREAD

**Issue by joker77877** [antfu/unplugin-vue-components#219]
When using this plugin, I can't normally debugging in chrome. For example, I can't set breakpoint at below position:
![image](https://user-images.githubusercontent.com/62642184/142180188-4a88578e-00cf-4fbc-8682-989e44fec2c5.png)

reproduce repositiory: https://github.com/joker77877/vue3-ts-eslint/blob/master/src/components/HelloWorld.vue

**Comment by mrgulc** [comment]
yes! me too

**Comment by joker77877** [comment]
Can you solve it quickly? It will affect code debugging.

**Comment by gaopengpian** [comment]
me too

**Comment by smallp** [comment]
Same issue.

**Comment by gaopengpian** [comment]
please fix it

**Comment by smallp** [comment]
I have solved this problem. Add `build: {    sourcemap: true,  },` to vite.config.js and it should work fine. This plugin depends this config to generate sourcemap.

**Comment by Djaler** [comment]
@smallp this doesn't help me. I have latest version of vite (2.7.13), vite-plugin-vue2 (1.9.3) and unplugin-vue-components (0.17.14), build.sourcemap is set to true in vite config, but I still get incorrect source map:
![image](https://user-images.githubusercontent.com/7964583/151319560-609645e8-5a84-4642-9178-5c07fa2701fd.png)
There should be imports of vuetify components (v-container, v-row, etc)

**Comment by joker77877** [comment]
it didn't help me either.

**Comment by smallp** [comment]
> @smallp this doesn't help me. I have latest version of vite (2.7.13), vite-plugin-vue2 (1.9.3) and unplugin-vue-components (0.17.14), build.sourcemap is set to true in vite config, but I still get incorrect source map: ![image](https://user-images.githubusercontent.com/7964583/151319560-609645e8-5a84-4642-9178-5c07fa2701fd.png) There should be imports of vuetify components (v-container, v-row, etc)

You dont have to set build.sourcemap now. Try again and give me a sample if it still cannot work.

**Comment by smallp** [comment]
@Djaler Yes, the source map of build result is wrong. But you can use `npm run dev`. That works fine.

**Comment by Djaler** [comment]
Sure, but I need correct maps in production, because of Sentry

**Comment by smallp** [comment]
> Sure, but I need correct maps in production, because of Sentry

Sorry, I'm not able to help you. I dont know the process of building.

**Comment by zcmk123** [comment]
> Sure, but I need correct maps in production, because of Sentry

can you set breakpoint in your project?
i clone your project, then click the button, console out put "could not load the sourcemap 404"
i'm using vue3 + vite with this plugin, upgrade plugin to latest version. still can not set breakpoint in devtools

**Comment by LiuQuanYou** [comment]
me too

**Comment by VincentDream** [comment]
me too

**Comment by Mehdi-Hp** [comment]
Well, I spent hours and now I know `unplugin-vue-components` is causing my issue and there is no solution

**Comment by rezelute** [comment]
We spent ages trying to figure out the reason debugging was not working in our new project and also finally made this same discovery today. It seems that Composition API code debugging works but not Options API. Is there any potential resolution for this? Thank you.

**Comment by jeddi** [comment]
We have the same problem + our tests coverage is wrong!

**Comment by LoicMonard** [comment]
Same problem here, any updates ?

**Comment by nlvedwin** [comment]
i got same problem

**Comment by h4091** [comment]
```
      {
        ...Components({
          resolvers: [VantResolver()],
        }),
        enforce: 'post',  // Add
      },
```
This solved my problem

**Comment by jeddi** [comment]
Hello i have found a work-around that works for me: 
add this to vue file:
```
<script setup>
// work-around https://github.com/antfu/unplugin-vue-components/issues/219
</script>
```

**Comment by LoicMonard** [comment]
Yeah the `script setup` is working but it's a problem if your whole codebase isn't based on `script setup`

**Comment by superchangme** [comment]
does any one resolve this problem?

**Comment by superchangme** [comment]
i use <script lang="jsx setup> will apprear this problem

**Comment by gaopengpian** [comment]
I fixed this problem a year ago, but I’m not sure which of the following changes took effect.
1.  remove plugin ： vite-plugin-vue-setup-extend
2. vite-plugin-compression ： compressPlugin add config ：verbose: false
3. "unplugin-auto-import": "0.5.3"   update to 0.5.11

**Comment by galaa2011** [comment]
so? how to reslove?

**Comment by Code-terminal** [comment]
how to reslove? this error afftect developing sence a lot.

**Comment by ys3322** [comment]
Using this plugin, if the **Vue version is greater than 3** and **the options syntax is used in the component**, the bug of the debugger offset will be reappear！！！

**Comment by LonelySunset** [comment]
> 使用这个插件，如果**vue版本大于3**，**组件中使用了options语法**，调试器偏移的bug又会出现！！！！

me too

**Comment by LonelySunset** [comment]
> Using this plugin, if the **Vue version is greater than 3** and **the options syntax is used in the component**, the bug of the debugger offset will be reappear！！！

me too

**Comment by MrRabbit1993** [comment]
> Using this plugin, if the **Vue version is greater than 3** and **the options syntax is used in the component**, the bug of the debugger offset will be reappear！！！

yes, i got same problem.
this problem occurs in jsx or tsx.

unplugin-vue-components": "^0.25.0"

**Comment by al123xiaaaa** [comment]
What a terrible bug! It completely undermines the Vue 3 DX, wasting a significant amount of time for those using this library in enterprise projects and forcing people into a 996 work schedule. This bug has persisted for a year and a half now...

**Comment by shanguoteng** [comment]
使用<script lang="jsx">的时候会有这个问题

**Comment by elephantasticio** [comment]
Hi,

Also affected by this bug. Is there an official workaround or fix ?

**Comment by anpr** [comment]
We have the same problem

**Comment by kevinkosterr** [comment]
Issue seems to still be relevant.
